src/Workspaces/Core/Portable/EmbeddedLanguages/DateAndTime/LanguageServices/DateAndTimePatternDetector.cs

```diff
@@ -133,32 +137,41 @@ private static bool IsMethodArgument(SyntaxToken token, ISyntaxFacts syntaxFacts
 
         public bool IsDateAndTimeToken(SyntaxToken token, CancellationToken cancellationToken)
         {
-            if (!IsPossiblyDateAndTimeToken(
+            if (IsPossiblyDateAndTimeTokenInInvocation(
                     token, _info.SyntaxFacts,
                     out var argumentNode, out var invocationOrCreation))
             {
-                return false;
-            }
+                // if we couldn't determine the arg name or arg index, can't proceed.
+                var (argName, argIndex) = GetArgumentNameOrIndex(argumentNode);
+                if (argName == null && argIndex == null)
+                    return false;
+
+                // If we had a specified arg name and it isn't 'format', then it's not a DateTime
+                // 'format' param we care about.
+                if (argName != null && argName != FormatName)
+                    return false;
+
+                var symbolInfo = _semanticModel.GetSymbolInfo(invocationOrCreation, cancellationToken);
+                var method = symbolInfo.Symbol;
+                if (TryAnalyzeInvocation(method, argName, argIndex))
+                    return true;
 
-            // if we couldn't determine the arg name or arg index, can't proceed.
-            var (argName, argIndex) = GetArgumentNameOrIndex(argumentNode);
-            if (argName == null && argIndex == null)
-                return false;
+                foreach (var candidate in symbolInfo.CandidateSymbols)
+                {
+                    if (TryAnalyzeInvocation(candidate, argName, argIndex))
+                        return true;
+                }
 
-            // If we had a specified arg name and it isn't 'format', then it's not a DateTime
-            // 'format' param we care about.
-            if (argName != null && argName != FormatName)
                 return false;
+            }
 
-            var symbolInfo = _semanticModel.GetSymbolInfo(invocationOrCreation, cancellationToken);
-            var method = symbolInfo.Symbol;
-            if (TryAnalyzeInvocation(method, argName, argIndex))
-                return true;
-
-            foreach (var candidate in symbolInfo.CandidateSymbols)
+            if (IsPossiblyDateAndTimeTokenInStringInterpolation(token, _info.SyntaxFacts))
             {
-                if (TryAnalyzeInvocation(candidate, argName, argIndex))
-                    return true;
+                var type = _semanticModel.GetTypeInfo(
+                    _info.SyntaxFacts.GetExpressionOfInterpolation(token.Parent!.Parent),
```

Comment:
could IsPossiblyDateAndTimeTokenInStringInterpolation return teh values you need so you don't need to `!` things.